android/src/main/java/io/customer/reactnative/sdk/messagingpush/PermissionStatus.kt

```diff
@@ -0,0 +1,9 @@
+package io.customer.reactnative.sdk.messagingpush
+
+/**
+ * Enum class for wrapping status of Android app permissions.
+ */
+enum class PermissionStatus {
```

Comment:
I made a [suggestion to the push permissions iOS PR](https://github.com/customerio/customerio-reactnative/pull/101#discussion_r1120711206) for the permission status we return. 

If that suggestion goes through, we can make the Android and iOS PermissionStatus identical. This has some nice benefits. 

1. We can modify `CustomerIOTracking.getPushPermissionStatus` RN function to return a strongly typed data type instead of the String it does today. 
2. We can add a "not asked" option for Android. I like that idea because I find value in [modifying this function](https://github.com/customerio/customerio-reactnative/pull/104/files#diff-74884e5e24c17161d56be01a332a64aeabd1f5bbafefab2bbd409befbb72fdd6R64) with saying, 

```kt
if (Build.VERSION.SDK_INT < BUILD_VERSION_CODE_TIRAMISU) { return .notAsked }
```

Instead of .Granted as we return right now in the PR.

Reply:
1. Sounds good. If we update it for iOS, I'll update it for Android too to match the expectations.
2. I don't see any benefit of adding additional `notAsked` status. It is rather confusing for me to handle more values that are uncommon. I would instead be happier with simple `Granted`, `Denied` or `Blocked` kind of permissions.